stargate/crates/query-planner/src/context.rs

```diff
@@ -214,35 +217,35 @@ impl<'q> QueryPlanningContext<'q> {
         }
 
         let return_type = self.names_to_types.get(field_def.field_type.as_name());
-        let field_type_is_not_composite =
-            return_type.is_none() || !return_type.unwrap().is_composite_type();
 
-        if field_type_is_not_composite {
-            return vec![];
-        }
+        if let Some(return_type) = return_type {
```

Comment:
the logic here is hard to read and would benefit from being broken into some smaller functions

Reply:
Tracking this in https://github.com/apollographql/federation/issues/477